api/bootstrap.proto

```diff
@@ -99,6 +99,54 @@ message StatsSink {
   google.protobuf.Struct config = 2;
 }
 
+// Designates a tag to strip from the tag extracted name and provide as a named
+// tag value for all statistics. This will only occur if a any part of the name
+// matches the regex provided with one or more capture groups.
+message TagSpecifier {
+  // Attaches an identifier to the tag values to identify the tag being in the
+  // sink. Envoy has a set of default names and regexes to extract dynamic
+  // portions of existing stats, which can be found in
+  // source/common/config/well_known_names.h in the Envoy repository. If a
+  // tag_name is provided in the config with an empty regex, Envoy will attempt
+  // to find that name in its set of defaults and use the accompanying regex.
+  // Note: if any default tags are specified twice, the config will be
+  // considered invalid.
+  string tag_name = 1;
+
+  // The first capture group identifies the portion of the name to remove. The
+  // second capture group (which will normally be nested inside the first)
+  // will designate the value of the tag for the statistic. If no second 
+  // capture group is provided, the first will also be used to set the value of
+  // the tag. All other capture groups will be ignored.
+  //
+  // For example:
+  //
+  // stat name: "cluster.foo_cluster.upstream_rq_timeout"
+  //
+  // tag name: "envoy.cluster_name" (one of many default stat regexes provided
+  // in Envoy)
+  //
+  // regex: "^cluster\.((.+?)\.)"
+  //
+  // Notice the regex will remove "foo_cluster.", but the tag value will be
+  // "foo_cluster".
```

Comment:
What happens to the `upstream_rq_timeout` portion above? I think another worked example would also help clarify.